src/renderer/atlas/AtlasEngine.cpp

```diff
@@ -360,6 +360,36 @@ try
         _api.dirtyRect = til::rect{ 0, _api.invalidatedRows.x, _api.cellCount.x, _api.invalidatedRows.y };
     }
 
+    // This is an important block of code for our TileHashMap.
+    // We only process glyphs within the dirtyRect, but glyphs outside of the
+    // dirtyRect are still in use and shouldn't be discarded. This is critical
+    // if someone uses a tool like tmux to split the terminal horizontally.
+    // If they then print a lot of Unicode text on just one side, we have to
+    // ensure that the (for example) plain ASCII glyphs on the other half of the
+    // viewport are still retained. This bit of code "refreshes" those glyphs and
+    // brings them to the front of the LRU queue to prevent them from being reused.

```

Comment:
how expensive is this refresh?

Reply:
would this be easier if we did not track dirty rects at all, and just did the whole screen every time?

Reply:
We need to avoid measuring glyphs with DirectWrite as much as possible, due to its insufficient font fallback performance. While not tracking dirty rects improves performance for ASCII ¹, it would make eastern scripts very unpleasant to use.

Brahmi script as used in India for instance runs at approx. 100k-150k code points per second on my fairly powerful workstation, compared to approx. 9M code points per second for Latin scripts. While 100k code points per second _sounds_ much, that's only ~30 FPS at 120x30 with my CPU running at 5 GHz. This means we have to cache DirectWrite results as much as possible and thus requires this workaround.

¹ Approx. 1.5x with DirectWrite, approx. 2-3x with Uniscribe.